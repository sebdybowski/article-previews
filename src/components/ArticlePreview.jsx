import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

export const ArticlePreview = ({ article }) => <article className="card mb-3">
    <div className="row no-gutters">
        <div className="col-xs-12 col-md-5 col-lg-4">
            <img src={article.image || 'https://placehold.co/280x180?text=Image+not+found...'} className="card-img" alt={`${article.title} image`} />
        </div>
        <div className="col-xs-12 col-md-7 col-lg-8">
            <div className="card-body">
                <div className="row">
                    <div className="col-8">
                        <h5 className="card-title">{article.title}</h5>
                    </div>
                    <div className="col-4">
                        <small className="text-muted float-right">{article.date}</small>
                    </div>
                </div>
                <p className="card-text d-none d-md-block">{_.truncate(article.preamble, { length: 120 })}</p>
            </div>
        </div>
    </div>
</article>;

ArticlePreview.propTypes = {
    article: PropTypes.shape({
        category: PropTypes.string,
        date: PropTypes.string,
        id: PropTypes.number,
        image: PropTypes.string,
        preamble: PropTypes.string,
        title: PropTypes.string,
    }),
};

ArticlePreview.defaultProps = {
    article: {},
};
