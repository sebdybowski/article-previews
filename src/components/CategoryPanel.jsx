import React from 'react';
import PropTypes from 'prop-types';

export const CategoryPanel = () => <ul className="nav flex-column">
    <li className="nav-item">
        <a className="nav-link active" href="#">Sports</a>
    </li>
    <li className="nav-item">
        <a className="nav-link" href="#">Fashion</a>
    </li>
</ul>;