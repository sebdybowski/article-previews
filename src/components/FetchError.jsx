import React from 'react';
import PropTypes from 'prop-types';
import { resolveErrorCategory } from '../utils/errors';

export const FetchError = ({ errorData }) =>
    <div className="alert alert-danger" role="alert">
        <p>
            { 'We are sorry, but requests from category ' }
            <span className="font-weight-bold">{resolveErrorCategory(errorData)}</span>
            { ' were not loaded due to ' }
            <span className="font-weight-bold">{errorData.response.status}</span>
            { ' error. This bug is temporary, please: ' }
        </p>
        <button onClick={() => window.location.reload()} type="button" className="btn btn-primary btn-block">Reload the page</button>
    </div>;

FetchError.propTypes = {
    errorData: PropTypes.object,
};