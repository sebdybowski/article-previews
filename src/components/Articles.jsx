import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { ArticlePreview } from './ArticlePreview';

export const Articles = ({ articles }) => _.map(articles, article =>
    <ArticlePreview article={article} key={article.id}/>);

Articles.propTypes = {
    articles: PropTypes.arrayOf(PropTypes.shape({
        category: PropTypes.string,
        date: PropTypes.string,
        dateSortable: PropTypes.number,
        id: PropTypes.number,
        image: PropTypes.string,
        preamble: PropTypes.string,
        title: PropTypes.string,
    })),
};