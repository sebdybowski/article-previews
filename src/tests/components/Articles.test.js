import React from 'react';
import { shallow } from 'enzyme';
import { Articles } from '../../components/Articles';

const mockProps = {
    articles: [
        {
            category: 'fashion',
            date: '7. februar 2019',
            dateSortable: 20190207,
            id: 986521,
            image: 'https://placeimg.com/280/200/any',
            preamble: 'Det finnes mange skrevne og uskrevne regler som beskriver hva en smoking egentlig er. Vi har tatt en titt på hvilke du bør følge, og hvilke du kan bryte.',
            title: 'Bør du følge smokingreglene?',
        },
        {
            category: 'fashion',
            date: '5. februar 2019',
            dateSortable: 20190205,
            id: 157489,
            image: 'https://placeimg.com/300/180/arch',
            preamble: 'Kong Harald (81) får moteskryt for spenstig skovalg.',
            title: 'Kongen stilte i Moon Boots: – Helt konge!',
        },
    ]
};

describe('Articles', () => {
    it('should render correctly with default props', () => {
        const component = shallow(<Articles {...mockProps} />);
        expect(component).toMatchSnapshot();
    });
});