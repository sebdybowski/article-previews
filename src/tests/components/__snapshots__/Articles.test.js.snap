// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles should render correctly with default props 1`] = `
Array [
  <ArticlePreview
    article={
      Object {
        "category": "fashion",
        "date": "7. februar 2019",
        "dateSortable": 20190207,
        "id": 986521,
        "image": "https://placeimg.com/280/200/any",
        "preamble": "Det finnes mange skrevne og uskrevne regler som beskriver hva en smoking egentlig er. Vi har tatt en titt på hvilke du bør følge, og hvilke du kan bryte.",
        "title": "Bør du følge smokingreglene?",
      }
    }
    key="986521"
  />,
  <ArticlePreview
    article={
      Object {
        "category": "fashion",
        "date": "5. februar 2019",
        "dateSortable": 20190205,
        "id": 157489,
        "image": "https://placeimg.com/300/180/arch",
        "preamble": "Kong Harald (81) får moteskryt for spenstig skovalg.",
        "title": "Kongen stilte i Moon Boots: – Helt konge!",
      }
    }
    key="157489"
  />,
]
`;
